src/main/java/org/elasticsearch/index/query/IdsQueryBuilder.java

```diff
@@ -159,11 +149,13 @@ public Query toQuery(QueryParseContext parseContext) throws IOException, QueryPa
             return Queries.newMatchNoDocsQuery();
         }
 
-        Collection<String> typesForQuery = this.types;
-        if (typesForQuery == null || typesForQuery.isEmpty()) {
+        Collection<String> typesForQuery;
+        if (types == null || types.length == 0) {
             typesForQuery = parseContext.queryTypes();
-        } else if (typesForQuery.size() == 1 && Iterables.getFirst(typesForQuery, null).equals("_all")) {
+        } else if (types.length == 1 && MetaData.ALL.equals(types[0])) {
             typesForQuery = parseContext.mapperService().types();
+        } else {
+            typesForQuery = Sets.newHashSet(types);
         }
 
         TermsQuery query = new TermsQuery(UidFieldMapper.NAME, Uid.createTypeUids(typesForQuery, ids));
```

Comment:
I sent #11263 to address this upstream, the method can easily accept a Collection instead of a List